integration_test/src/test_runner/test_generator_util.rs

```diff
@@ -0,0 +1,168 @@
+        expected_sql_type.push(sql_type);
+
+        let bson_type = get_column_attribute(&stmt, i, Desc::TypeName, &STRING_VAL)?;
+        expected_bson_type.push(bson_type);
+
+        let precision = get_column_attribute(&stmt, i, Desc::Precision, &NUMBER_VAL)?;
+        expected_precision.push(precision);
+
+        let scale = get_column_attribute(&stmt, i, Desc::Scale, &NUMBER_VAL)?;
+        expected_scale.push(scale);
+
+        let nullability = get_column_attribute(&stmt, i, Desc::Nullable, &NUMBER_VAL)?;
+        expected_nullability.push(nullability);
+    }
+
+    // 2. Get result set data
+    let mut expected_result: Vec<Vec<Value>> = vec![];
+
+    while fetch_row(&stmt)? {
+        let mut row: Vec<Value> = vec![];
+        for i in 0..(column_count) {
+            let expected_data_type = get_expected_data_type(expected_sql_type.get(i).unwrap());
+            let field = get_data(&stmt, i as USmallInt, expected_data_type)?;
+            row.push(field);
+        }
+        expected_result.push(row);
+    }
+
+    // 3. Create new TestEntry with all the data
+    let test_entry = TestEntry {
+        description: entry.description.clone(),
+        db: entry.db.clone(),
+        test_definition: entry.test_definition.clone(),
+        expected_result: Some(expected_result),
+        skip_reason: None,
+        ordered: entry.ordered,
+        expected_catalog_name: Some(expected_catalog_name),
+        expected_case_sensitive: Some(expected_case_sensitive),
+        expected_column_name: Some(expected_column_name),
+        expected_display_size: Some(expected_display_size),
+        expected_length: Some(expected_length),
+        expected_is_searchable: Some(expected_is_searchable),
+        expected_is_unsigned: Some(expected_is_unsigned),
+        expected_sql_type: Some(expected_sql_type),
+        expected_bson_type: Some(expected_bson_type),
+        expected_precision: Some(expected_precision),
+        expected_scale: Some(expected_scale),
+        expected_nullability: Some(expected_nullability),
+    };
+
+    // 4. Write the TestEntry to a file
+    let now = std::time::SystemTime::now()
+        .duration_since(std::time::UNIX_EPOCH)
+        .unwrap()
+        .as_millis();
+    let desc = entry.description.clone().replace(" ", "_");
+    let file_name = format!("{}-{}.yml", desc, now);
+
+    let writer = std::fs::OpenOptions::new()
+        .write(true)
+        .create(true)
+        .open(format!("{}/{}", GENERATED_TEST_DIR, file_name))
+        .expect("could not open or create test file");
+
+    serde_yaml::to_writer(writer, &test_entry).map_err(|err| Error::Yaml(err.to_string()))
+}
+
+// Get the expected CDataType for the provided sql_type.
+fn get_expected_data_type(sql_type: &Value) -> CDataType {
+    match sql_type {
+        Value::Number(n) => {
+            let sdt = SqlDataType(n.as_i64().unwrap() as i16);
+            match sdt {
+                SqlDataType::UNKNOWN_TYPE => CDataType::Default,
```

Comment:
We don't support `CDataType::Default`, of course, but I wasn't sure what to put here. `Char`? Something else?

Reply:
Are we able to convert all the UNKOWN_TYPEs to char?  This was something I was not sure about when writing the test runner.  `Char` would be fine, similar to how in the JDBC test harness we have the option of using getString().

Reply:
I'm not sure, maybe @pmeredit can weigh in on this one

Reply:
I think `Char` is fine.